### PR TITLE
[wip] Make logins work

### PR DESCRIPTION
**Pull request by mozbhearsum**
This patch fixes things up enough to make logins work, but it's not ready for review or landing. At the very least, the following needs to be addressed:
* Stop hardcoding the Firefox CI production cluster and client information - this needs to be configured in the environment somewhere so that localdev/staging/production can all use different values.
* Update "Manage Credentials" link to point at whatever makes sense in new TC land

And while not strictly necessary for this PR, I do think it would be good to make internal artifact downloads work at the same time. Right now, even when you're logged in they don't work because they use a plain `<a>`, which means there's no `Authorization` header sent. I'm pretty sure it's not possible to send that header with an `<a>` link, so we may need to stuff it into a cookie, or use an alternative way of making the request to the backend.

### CODE REVIEW

**Comment by mozbhearsum** [comment]
There's still a couple of things to do here, but I wouldn't mind some feedback in the meantime, @rail. I think it's mostly straightforward, except for the JS config stuff, which is a bit nasty (I haven't found a better way to handle it).

File: api/src/tooltool_api/static/tcauth.js [hunk 1]
```diff
@@ -90,16 +95,8 @@ window.tcauth.setup = function(service, default_service_url) {
...
-        $.ajax({
-            url: url + '/auth0/login',
-            error: function(xhr, status, error) {
-            },
-            success: function(redirect_url, status, xhr) {
-                window.location = redirect_url;
-            }
-
-        });
+        var url = config.taskclusterRootUrl + '/login/oauth/authorize?client_id=' + config.clientId + '&releng-tooltool-localdev&redirect_uri=' + config.redirectUri + '&response_type=code&scope=project%3Areleng%3Aservices%2Ftooltool%2F*&state=5&expires=5%20minutes';
```
Comment (rail) [inline api/src/tooltool_api/static/tcauth.js]: `releng-tooltool-localdev` is hardcoded, you probably forgot to remove it. `scope` looks like a good candidate to move to the configs.

**Comment by rail** [comment]
Hmm, interesting. When I refresh a page after I login, I get an alert and a 500 from https://firefox-ci-tc.services.mozilla.com/login/oauth/credentials

**Comment by rail** [comment]
BTW, `api/src/tooltool_api/static/index.html` still has a reference to taskcluster.net.

File: api/src/tooltool_api/static/tcauth.js (hunk 1 above)
Comment (mozbhearsum) [reply api/src/tooltool_api/static/tcauth.js]: Oops, yes! It's not breaking anything, but it doesn't need to be there.

**Comment by mozbhearsum** [comment]
> BTW, `api/src/tooltool_api/static/index.html` still has a reference to taskcluster.net.

Yeah, that's the credentials link that I need to update. Will do.



> Hmm, interesting. When I refresh a page after I login, I get an alert and a 500 from https://firefox-ci-tc.services.mozilla.com/login/oauth/credentials:
> `{"name":"InputError","message":"Could not generate credentials for this access token"}`

Hmmm....I'll look into this. That's not ideal.

**Comment by mozbhearsum** [comment]
> And while not strictly necessary for this PR, I do think it would be good to make internal artifact downloads work at the same time. Right now, even when you're logged in they don't work because they use a plain <a>, which means there's no Authorization header sent. I'm pretty sure it's not possible to send that header with an <a> link, so we may need to stuff it into a cookie, or use an alternative way of making the request to the backend.

I'm giving up on this part. Users can already download files through the command line client, so it's not that important to be able to do through the UI. If we ever rebuild the tooltool UI we should probably do it a way that makes this easier.

**Comment by mozbhearsum** [comment]
I think this is as good as its going to get for now. Logins work, downloads of internal artifacts don't, and I fixed the error on refresh.

**Pull request merged**